docs/grammar/SolidityParser.g4

```diff
@@ -247,6 +249,11 @@ structMember: type=typeName name=identifier Semicolon;
  * Definition of an enum. Can occur at top-level within a source unit or within a contract, library or interface.
  */
 enumDefinition:	Enum name=identifier LBrace enumValues+=identifier (Comma enumValues+=identifier)* RBrace;
+/**
+ * Definition of a user defined value type. Can occur at top-level within a source unit or within a contract, library or interface.
+ */
+userDefinedValueTypeDefinition:
+	Type name=identifier Is elementaryTypeName[true] Semicolon;
```

Comment:
```suggestion
	Type name=identifier Is elementaryTypeName[false] Semicolon;
```
I would not allow `address payable` here since it serves no purpose and makes the grammar more complicated.

Reply:
Do you want to disallow `type MyPayableAddress is address payable`? This is currently allowed. I think there is no good reason to disallow it.

Reply:
I'd say it should be allowed.

Reply:
Ok!